devito/finite_differences/coefficients.py

```diff
@@ -243,8 +243,10 @@ def generate_subs(deriv_order, function, index):
 
         subs = {}
 
+        mapper = frozendict({dim: index})
```

Comment:
I still don't get why this all piece of code exist (`default_rule`). In `finite_difference.py` the indices are computed already should only need to add the coefficients to it (but maybe I'm missing something @rhodrin i the loop).

Reply:
To my understanding, the process is: Stencils evaluated but with `w(x+h_x)` etc instead of standard coefficients -> Fill in the coefficients for which no substitutions have been given -> Fill in the coefficients for which substitutions have been given. I presume the indices are 'baked in' and discarded once step one has been carried out? As far as I can tell, they are not readily available at the start of the second step